### ISSUE TITLE
Antox disconnects when put in background

### ISSUE THREAD

**Issue by Answer-Man** [Antox/Antox#480]
As said in the title, when I open Antox and then leave it running in background it automatically disconnects, I have to be constantly on the app to chat. The same happens when the screen goes on standby.

**Comment by licaon-kter** [comment]
Device? Android version? Disabled battery optimization for the app?

**Comment by Answer-Man** [comment]
Device: Samsung Galaxy S6 Edge.
Android : 6.0.1
General Battery Optimization was indeed activated and I was connected over 4G.
The issue seems to happens only when using both 4G and battery optimization together.
Weirdly enough other apps of instant messaging continue to work in these same conditions.

**Comment by toxtox** [comment]
I have Android 6.0.1 too. I played around with **_"Battery Optimization"_** on/off for Antox but I could not see any effect concerning connection problems.
I guess you have enabled "**_Restrict background data_**". This option disables mobile data (4G) or throttles down the allowed bandwidth when screen is off. But Antox needs a very high constant idle bandwidth (I measured 1,5MB/5Minutes). If the bandwidth requirement is not fulfilled Antox disconnects.
Your other instant messaging apps seem to work because these apps reconnect immediately after you switch on your screen.
But **_Antox_** does NOT reconnect after the second bandwidth interruption if the interruptions take more than several minutes https://github.com/Antox/Antox/issues/477 . This is because of an very old and annoying bug in the **_c-toxcore_** library. https://github.com/TokTok/c-toxcore/issues/240
But this could be fixed in Antox with a **workaroud**:
This is how the **workaround** could look like:

- Antox client goes offline because of interruption of internet connection
- Antox starts doing checks continuously or every 5 seconds in background until an internet connection is available again. (I guess there is an Android for checking for an available internet connection)
- Then Antox programmatically logs out and logs in again. => Antox and contacts/friends are shown as online again

**But to find someone who could fix the bug in _c-toxcore_ or to write the workaround for Antox seems to be very difficult.**

**Comment by Answer-Man** [comment]
Thank you for your precise and complete response. 
I hope someone can actually fix this.

**Comment by physkets** [comment]
My friend too has run into this issue on Android 7.1. The Antox client on the phone goes offline when the phone's screen is turned off. I comes back up only when the phone is used again. This makes it almost unusable.

**Comment by zoff99** [comment]
@physkets @Answer-Man what version of Antox are you using? where did you install it from?
do you have battery optimizations turned on or off for Antox?

**Comment by uihsnv** [comment]
@zoff99 

* Version 0.25.495
* It is installed from the Antox F-Droid repository: 
* The "Experimental Battery Saving Mode" is turned **OFF**
* This is the only client I've tried. Do you want me tor try something specific?

**Comment by zoff99** [comment]
@uihsnv 
* do you have battery optimizations turned on or off for Antox?

with this i actually meant the android system settings

**Comment by physkets** [comment]
@zoff99 

* Version 0.25.495
* It is installed from the Antox F-Droid repository: 
* The "Experimental Battery Saving Mode" is turned **OFF**
* This is the only client I've tried. Do you want me tor try something specific?

**Comment by physkets** [comment]
@zoff99 Yep, battery optimization was turned on. I have now turned it off, and will try it out.

**Comment by physkets** [comment]
@zoff99 Okay, I tried it for a couple of days, and it works very well now. So that was the issue. Can't you request an Android 'permission' to disable 'battery optimisation' for the app?
